src/nvim/state.c

```diff
@@ -160,6 +160,8 @@ char *get_mode(void)
     buf[0] = 'n';
     if (finish_op) {
       buf[1] = 'o';
+      // to able to detect force-linewise/blockwisecharacterwise operations
+      buf[2]=motion_force;
```

Comment:
You also need to change the allocation size from `3` to `4`, to ensure room for the NUL.